### PR TITLE
Makefile, CI: add checkov checks

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Add checkov checks for the policies that have high severity

Closes: #712
Signed-off-by: Hyeongju Johannes Lee <hyeongju.lee@intel.com>

### CODE REVIEW

**Comment by hj-johannes-lee** [comment]
@bart0sh I think so,, considering that the last merged pr didn't have any problem...?
What I don't understand is how my pr is related to Jenkins...

**Comment by hj-johannes-lee** [comment]
@mythi @bart0sh Can you take a look?
We have 5 polices that actually make failures.

Let me not use BC_K8S_XX form anymore, but rather CKV_K8S_XX form. (Idk why they made it so complicated..)
- Ensure container is not privileged ([CKV_K8S_16](https://docs.bridgecrew.io/docs/bc_k8s_15))
- Ensure containers do not run with AllowPrivilegeEscalation ([CKV_K8S_20](https://docs.bridgecrew.io/docs/bc_k8s_19))
- Minimize admission of root containers ([CKV_K8S_23](https://docs.bridgecrew.io/docs/bc_k8s_22))
- Ensure CAP_SYS_ADMIN Linux capability is not used ([CKV_K8S_39](https://docs.bridgecrew.io/docs/bc_k8s_36))
- Ensure containers do not share the host network namespace ([CKV_K8S_19](https://docs.bridgecrew.io/docs/bc_k8s_18))

If I am right, at this moment, we just skip the first four ones, right?

And, for the last one,,, what do I do?
It happens in
deployments/sgx_aesmd/base/kustomization.yaml
deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml

I guess (hostNetwork: true) is also.. intended, right? So, do we just skip this one too?

**Comment by bart0sh** [comment]
@hj-johannes-lee I'm getting this traceback when trying to run this check on Ubuntu 18.04:
```
$ make checkov
_work/venv/bin/pip3 install -U checkov
Requirement already up-to-date: checkov in ./_work/venv/lib/python3.6/site-packages
Requirement already up-to-date: docopt==0.6.2 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: urllib3==1.25.10 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: idna==2.8 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: junit-xml==1.8 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: packaging==20.4 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: jmespath==0.10.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: pyyaml==5.3.1 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: lark-parser==0.7.8 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: deep-merge==0.0.4 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: colorama==0.4.3 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: tqdm==4.49.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: dpath==1.5.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: update-checker==0.18.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: bc-python-hcl2>=0.3.11 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: tabulate==0.8.6 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: chardet==3.0.4 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: semantic-version==2.8.5 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: GitPython==3.1.7 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: boto3==1.12.43 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: six==1.15.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: termcolor==1.1.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: requests==2.22.0 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: gitdb==4.0.5 in ./_work/venv/lib/python3.6/site-packages (from checkov)
Requirement already up-to-date: pyparsing>=2.0.2 in ./_work/venv/lib/python3.6/site-packages (from packaging==20.4->checkov)
Requirement already up-to-date: s3transfer<0.4.0,>=0.3.0 in ./_work/venv/lib/python3.6/site-packages (from boto3==1.12.43->checkov)
Requirement already up-to-date: botocore<1.16.0,>=1.15.43 in ./_work/venv/lib/python3.6/site-packages (from boto3==1.12.43->checkov)
Requirement already up-to-date: certifi>=2017.4.17 in ./_work/venv/lib/python3.6/site-packages (from requests==2.22.0->checkov)
Requirement already up-to-date: smmap<4,>=3.0.1 in ./_work/venv/lib/python3.6/site-packages (from gitdb==4.0.5->checkov)
Requirement already up-to-date: python-dateutil<3.0.0,>=2.1 in ./_work/venv/lib/python3.6/site-packages (from botocore<1.16.0,>=1.15.43->boto3==1.12.43->checkov)
Requirement already up-to-date: docutils<0.16,>=0.10 in ./_work/venv/lib/python3.6/site-packages (from botocore<1.16.0,>=1.15.43->boto3==1.12.43->checkov)
_work/venv/bin/checkov \
	-c CKV_K8S_2,CKV_K8S_16,CKV_K8S_39,CKV_K8S_100,CKV_K8S_102,CKV_K8S_104,CKV_K8S_105,CKV_K8S_108,CKV_K8S_110,CKV_K8S_111,CKV_K8S_113,CKV_K8S_115,CKV_K8S_116,CKV_K8S_118,CKV_K8S_119,CKV_K8S_121,CKV_K8S_148,CKV_K8S_149,CKV_K8S_71,CKV_K8S_72,CKV_K8S_73,CKV_K8S_86,CKV_K8S_88,CKV_K8S_96,CKV_K8S_99,CKV_K8S_1,CKV_K8S_3,CKV_K8S_4,CKV_K8S_6,CKV_K8S_17,CKV_K8S_18,CKV_K8S_19,CKV_K8S_20,CKV_K8S_23,CKV_K8S_27,CKV_K8S_5,CKV_K8S_106,CKV_K8S_107,CKV_K8S_112,CKV_K8S_117,CKV_K8S_138,CKV_K8S_150,CKV_K8S_74,CKV_K8S_75,CKV_K8S_78,CKV_K8S_79,CKV_K8S_80,CKV_K8S_85,CKV_K8S_91,CKV_K8S_95,CKV_K8S_97,CKV_K8S_49,CKV_K8S_6,CKV_K8S_79 \
	--quiet --compact \
	--soft-fail-on CKV_K8S_16,CKV_K8S_19,CKV_K8S_39,CKV_K8S_20,CKV_K8S_23 \
	--directory ./deployments \
	--framework kustomize
Traceback (most recent call last):
  File "_work/venv/bin/checkov", line 2, in <module>
    from checkov.main import run
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/main.py", line 16, in <module>
    from checkov.common.util.docs_generator import print_checks
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/common/util/docs_generator.py", line 11, in <module>
    from checkov.serverless.registry import sls_registry
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/__init__.py", line 1, in <module>
    from checkov.serverless.checks.function import *
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/checks/__init__.py", line 3, in <module>
    from checkov.serverless.checks.function import *
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/checks/function/__init__.py", line 1, in <module>
    from checkov.serverless.checks.function.aws import *
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/checks/function/aws/AdminPolicyDocument.py", line 1, in <module>
    from checkov.serverless.checks.function.base_function_check import BaseFunctionCheck
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/checks/function/base_function_check.py", line 5, in <module>
    from checkov.serverless.checks.function.registry import function_registry
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/checks/function/registry.py", line 1, in <module>
    from checkov.serverless.base_registry import ServerlessRegistry
  File "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/lib/python3.6/site-packages/checkov/serverless/base_registry.py", line 2, in <module>
    from dataclasses import dataclass
ModuleNotFoundError: No module named 'dataclasses'
Makefile:95: recipe for target 'checkov' failed
make: *** [checkov] Error 1
```

Any suggestions how to proceed?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -142,3 +142,12 @@ jobs:
...
           make test-with-kind
+
+  checkov:
+    name: Checkov Checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run checkov checks
+        run: |
+          make checkov
```
Comment (mythi) [inline .github/workflows/ci.yaml:153]: can we use github actions cache to store venv installation?

File: Makefile [hunk 2]
```diff
@@ -88,6 +91,20 @@ envtest:
...
+		--quiet --compact \
+		--soft-fail-on ${CHECKOV_SOFT_FAIL_ON} \
+		--directory ./deployments \
+		--framework kustomize
+	_work/venv/bin/checkov \
+		-c ${CHECKOV_HIGH_SEVERITY},${CHECKOV_MEDIUM_SEVERITY} \
+		--quiet --compact \
+		--soft-fail-on ${CHECKOV_SOFT_FAIL_ON} \
+		--directory ./ --skip-path ./deployments 
+
```
Comment (mythi) [inline Makefile:107]: what's this?

**Review by mythi — CHANGES_REQUESTED** [review]

**Comment by hj-johannes-lee** [comment]
> 

Python 3.7 is minimum. Would I just put python and pip3 installing process in the target install-tools..?

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (hj-johannes-lee) [reply Makefile:107]: You said..
> Final thought: should we limit checks to deployments/ only?

But,, that's when we consider only the matter of the policies
- Ensure containers do not run with AllowPrivilegeEscalation ([CKV_K8S_20](https://docs.bridgecrew.io/docs/bc_k8s_19))
- Minimize admission of root containers ([CKV_K8S_23](https://docs.bridgecrew.io/docs/bc_k8s_22))
?
Wouldn't we need other checks for other directories?

File: Makefile [hunk 3]
```diff
@@ -88,6 +91,20 @@ envtest:
 lint:
 	@golangci-lint run --timeout 15m
 
+checkov: _work/venv/.stamp
+	_work/venv/bin/pip3 install -U checkov
+	_work/venv/bin/checkov \
+		-c ${CHECKOV_HIGH_SEVERITY},${CHECKOV_MEDIUM_SEVERITY} \
+		--quiet --compact \
+		--soft-fail-on ${CHECKOV_SOFT_FAIL_ON} \
+		--directory ./deployments \
```
[the remark this answers is not in the document]
Comment (hj-johannes-lee) [reply Makefile:100]: No, as I told, this project has very limited functionalities at this point... Can I submit an issue about this matter.!
If the directories we need to scan are those,, we need to just remove one by one..
--skip-path "blah_blah_1" -- skip-path blah_blah_2 ...

[the remark this answers is not in the document]
Comment (mythi) [reply Makefile:100]: "To skip files or directories, use the argument --skip-path, which can be specified multiple times. This argument accepts regular expressions for paths relative to the current working directory. "

**Comment by bart0sh** [comment]
@hj-johannes-lee With python 3.7 it crashes this way:
```
Building wheels for collected packages: termcolor, multidict, lark-parser, pyrsistent, cffi
  Running setup.py bdist_wheel for termcolor ... error
  Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/termcolor/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpkhwrxt3lpip-wheel- --python-tag cp37:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for termcolor
  Running setup.py clean for termcolor
  Running setup.py bdist_wheel for multidict ... error
  Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/multidict/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmppq3wdb2xpip-wheel- --python-tag cp37:
  *********************
  * Accelerated build *
  *********************
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for multidict
  Running setup.py clean for multidict
  Running setup.py bdist_wheel for lark-parser ... error
  Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/lark-parser/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmph3siljhspip-wheel- --python-tag cp37:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for lark-parser
  Running setup.py clean for lark-parser
  Running setup.py bdist_wheel for pyrsistent ... error
  Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/pyrsistent/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp5gh9jeaepip-wheel- --python-tag cp37:
  /usr/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'long_description_content_type'
    warnings.warn(msg)
  /usr/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'license_files'
    warnings.warn(msg)
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for pyrsistent
  Running setup.py clean for pyrsistent
  Running setup.py bdist_wheel for cffi ... error
  Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/cffi/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp3qg9vw54pip-wheel- --python-tag cp37:
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libffi', required by 'world', not found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libffi', required by 'world', not found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libffi', required by 'world', not found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libffi', required by 'world', not found
  Package libffi was not found in the pkg-config search path.
  Perhaps you should add the directory containing `libffi.pc'
  to the PKG_CONFIG_PATH environment variable
  Package 'libffi', required by 'world', not found
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for cffi
  Running setup.py clean for cffi
Failed to build termcolor multidict lark-parser pyrsistent cffi
Installing collected packages: termcolor, jmespath, deep-merge, six, python-dateutil, botocore, s3transfer, boto3, websocket-client, docker, cachetools, decorator, ply, jsonpath-ng, async-timeout, multidict, yarl, asynctest, frozenlist, aiosignal, aiohttp, configargparse, dockerfile-parse, networkx, cached-property, pyyaml, contextlib2, schema, policy-sentry, click-option-group, cloudsplaining, smmap, gitdb, GitPython, update-checker, toml, types-setuptools, packageurl-python, types-toml, setuptools, cyclonedx-python-lib, pycparser, cffi, pycares, aiodns, detect-secrets, lark-parser, bc-python-hcl2, tqdm, policyuniverse, argcomplete, colorama, tabulate, wcwidth, prettytable, junit-xml, pyrsistent, jsonschema, checkov
  Running setup.py install for termcolor ... done
  Running setup.py install for multidict ... error
    Complete output from command /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/multidict/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-k_mmnwz3-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/include/site/python3.7/multidict:
    *********************
    * Accelerated build *
    *********************
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/multidict
    copying multidict/__init__.py -> build/lib.linux-x86_64-3.7/multidict
    copying multidict/_multidict_base.py -> build/lib.linux-x86_64-3.7/multidict
    copying multidict/_abc.py -> build/lib.linux-x86_64-3.7/multidict
    copying multidict/_compat.py -> build/lib.linux-x86_64-3.7/multidict
    copying multidict/_multidict_py.py -> build/lib.linux-x86_64-3.7/multidict
    running egg_info
    writing multidict.egg-info/PKG-INFO
    writing dependency_links to multidict.egg-info/dependency_links.txt
    writing top-level names to multidict.egg-info/top_level.txt
    reading manifest file 'multidict.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files found matching 'multidict/_multidict.html'
    warning: no previously-included files found matching 'multidict/*.so'
    warning: no previously-included files found matching 'multidict/*.pyd'
    warning: no previously-included files found matching 'multidict/*.pyd'
    no previously-included directories found matching 'docs/_build'
    writing manifest file 'multidict.egg-info/SOURCES.txt'
    copying multidict/__init__.pyi -> build/lib.linux-x86_64-3.7/multidict
    copying multidict/py.typed -> build/lib.linux-x86_64-3.7/multidict
    running build_ext
    building 'multidict._multidict' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/multidict
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/include -I/usr/include/python3.7m -c multidict/_multidict.c -o build/temp.linux-x86_64-3.7/multidict/_multidict.o -O2 -std=c99 -Wall -Wsign-compare -Wconversion -fno-strict-aliasing -pedantic
    multidict/_multidict.c:1:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-cm1op1zy/multidict/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-k_mmnwz3-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/ed/git/intel-device-plugins-for-kubernetes/_work/venv/include/site/python3.7/multidict" failed with error code 1 in /tmp/pip-build-cm1op1zy/multidict/
Makefile:95: recipe for target 'checkov' failed
make: *** [checkov] Error 1
```

Any suggestions?

**Comment by bart0sh** [comment]
@hj-johannes-lee @mythi there are quite a bit of alternatives to checkov that may suit our goals better. [terrascan](https://github.com/accurics/terrascan) is one of those. It's written in go, installation and usage is quite straightforward:
```
$ terrascan scan --severity high -i kustomize -d ./deployments/dlb_plugin/


Violation Details -
    
	Description    :	Allowing hostPaths to mount to Pod arise the probability of getting access to the node's filesystem
	File           :	kustomization.yaml
	Line           :	1
	Severity       :	HIGH
	-----------------------------------------------------------------------
	
	Description    :	Prefer using secrets as files over secrets as environment variables
	File           :	kustomization.yaml
	Line           :	1
	Severity       :	HIGH
	-----------------------------------------------------------------------
	
	Description    :	Minimize Admission of Root Containers
	File           :	kustomization.yaml
	Line           :	1
	Severity       :	HIGH
	-----------------------------------------------------------------------
	
	Description    :	Containers Should Not Run with AllowPrivilegeEscalation
	File           :	kustomization.yaml
	Line           :	1
	Severity       :	HIGH
	-----------------------------------------------------------------------
	

Scan Summary -

	File/Folder         :	/home/ed/git/intel-device-plugins-for-kubernetes/deployments/dlb_plugin
	IaC Type            :	kustomize
	Scanned At          :	2022-02-11 10:07:42.923438532 +0000 UTC
	Policies Validated  :	9
	Violated Policies   :	4
	Low                 :	0
	Medium              :	0
	High                :	4

```

I'm not insisting on using it, just thinking out loud. Would it be better to use another tool instead of fighting with the one that doesn't work for us?

**Comment by hj-johannes-lee** [comment]
@bart0sh I also took a look of it! It looks good that there is an option for setting severity.
We do not need to set which checks we will apply manually with this tool.
But, we need to set directories manually like below.

terrascan scan ... -d ./deployments/dlb_plugin -d ./deployments/dsa_plugin -d ./deployments/ -d ...

Well, since we need to add only when there is a new plugin, it seems better to me.

@mythi What do you think.?

**Comment by bart0sh** [comment]
I'd propose to evaluate other similar tools before making decision. I'd prefer something written in go and build statically as it wouldn't require to install any dependencies and will be faster than something written in Python, but that's debatable.

**Comment by bart0sh** [comment]
I'd add that checkov may be 100 times slower than terrascan. I didn't measure, but checkov felt slow and terraform responded instantly.

**Comment by bart0sh** [comment]
Here is an analisys of the checkov, terrascan and two other tools: https://www.revolgy.com/insights/blog/complete-guide-for-picking-the-right-tool-for-terraform-security-code-analysis

File: Makefile [hunk 4]
```diff
@@ -88,6 +91,15 @@ envtest:
 lint:
 	@golangci-lint run --timeout 15m
 
+checkov: _work/venv/.stamp
```
Comment (bart0sh) [inline Makefile:94]: Do we really need sphinx packages that are installed by _work/venv/.stamp target?

Comment (hj-johannes-lee) [reply Makefile:94]: No, just I thought it's better to use what already exists then just putting "python3 -m venv ..."

**Comment by hj-johannes-lee** [comment]
@bart0sh If we continue with checkov, what about just using just --skip-check flag to skip checks that we do not need to care at this moment? Tthen we do not need to manually specify which checks we will run.

**Comment by bart0sh** [comment]
> @bart0sh If we continue with checkov, what about just using just --skip-check flag to skip checks that we do not need to care at this moment? Tthen we do not need to manually specify which checks we will run.

That can work, but it will require some maintenance too. When new checks are added to checkov we'll need to update the list of skipped checks as most probably they'll be failing. This is ok, but if you add inability to set severity it can become a burden.

**Comment by bart0sh** [comment]
@hj-johannes-lee 
> Well, since we need to add only when there is a new plugin

would something like this or similar help?
```
ls deployments/*/kustomization.yaml | while read f ; do terrascan scan -i kustomize --severity high -d $(dirname $f) ; done
```

**Pull request closed without merging**